src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java

```diff
@@ -681,62 +709,32 @@ public RequestSerializer(String resourceId, String requestBody, String contentTy
 
         @Override
         protected void serializeRequest(ServiceRequest request, SOAPElement soapRequest, SOAPEnvelope envelope) throws SOAPException {
-            writeGetParametersToBody(request, soapRequest);
+            writeBodyContents(request, soapRequest);
             if (this.requestBody != null && !this.requestBody.isEmpty()) {
-                if (convertPost) {
-                    convertJsonToXmlAndWriteToSoap(this.requestBody, soapRequest);
-                } else {
-                    // send HTTP POST as an attachment
+                if (!convertPost) {
```

Comment:
Could be combined with the enclosing "if".

Reply:
Retained the separate if-statement for (in my view) better clarity; added else-block and more comments